web/server/api/endpoints/lineage.py

```diff
@@ -78,20 +75,17 @@ async def column_lineage(
     for i, node in enumerate(node.walk()):
         if i > 0:
             node_name = _get_table(node) or node.name
-            node_name = (
-                context.snapshots[node_name].name if node_name in context.snapshots else node_name
-            )
+            node_name = normalize_model_name(node_name, None)
             column_name = exp.to_column(node.name).name
         if column_name in graph.get(node_name, []):
             continue
 
-        node_fqn = normalize_model_name(node_name, context.default_catalog, context.config.dialect)
-        dialect = context.models[node_fqn].dialect if node_fqn in context.models else ""
+        dialect = context.models[node_name].dialect if node_name in context.models else ""
```

Comment:
context.get_model(...)

Reply:
how `context.get_model(...)` is different from `context.models[node_name]` since we still need to check if node is actually a model

Reply:
get_model normalizes stuff for you, so you don't need to call normalize_model_name manually

Reply:
yes, but what if node is not a table nor a model (cte, or  a random external model) ?
![Screenshot 2023-12-28 at 10 11 30 AM](https://github.com/TobikoData/sqlmesh/assets/5644753/e7482857-ddb1-46be-80dc-5be2842c158d)

Reply:
change get_table to get the expression and find the identifier / alias and call sql(identify=True)